mungegithub/mungers/matchers/comment/comment.go

```diff
@@ -0,0 +1,47 @@
+package comment
+
+import (
+	"time"
+
+	"github.com/google/go-github/github"
+)
+
+type Matcher interface {
+	Match(comment *github.IssueComment) bool
+}
+
+type And []Matcher
+
+func (a And) Match(comment *github.IssueComment) bool {
+	for _, matcher := range []Matcher(a) {
+		if !matcher.Match(comment) {
+			return false
+		}
+	}
+	return true
+}
+
+type Or []Matcher
+
+func (o Or) Match(comment *github.IssueComment) bool {
+	for _, matcher := range []Matcher(o) {
+		if matcher.Match(comment) {
+			return true
+		}
+	}
+	return false
+}
+
+type Not struct {
+	Matcher Matcher
```

Comment:
Actually, it's acceptable. :)

On Fri, Aug 12, 2016 at 2:14 PM, grodrigues3 <notifications@github.com>
wrote:

> In mungegithub/mungers/matchers/comment/comment.go
> <https://github.com/kubernetes/contrib/pull/1530#discussion_r74661409>:
>
> > +	return true
> > +}
> > +
> > +type Or []Matcher
> > +
> > +func (o Or) Match(comment *github.IssueComment) bool {
> > +	for _, matcher := range []Matcher(o) {
> > +		if matcher.Match(comment) {
> > +			return true
> > +		}
> > +	}
> > +	return false
> > +}
> > +
> > +type Not struct {
> > +	Matcher Matcher
>
> seems like it might be bad form to have the type and identifier be the same
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/contrib/pull/1530/files/064473f80ab1282a958ae38e367daecb91dc1649#r74661409>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAnglhWO2fuMIwj1GHokYikz8RD4b1mMks5qfOIfgaJpZM4JjXFP>
> .
>